sessionctx/variable/sysvar.go

```diff
@@ -1244,6 +1244,10 @@ var defaultSysVars = []*SysVar{
 		s.TMPTableSize = tidbOptInt64(val, DefTiDBTmpTableMaxSize)
 		return nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBTableCacheLease, Value: strconv.Itoa(DefTiDBTableCacheLease), Type: TypeUnsigned, MinValue: 1, MaxValue: 10, SetGlobal: func(vars *SessionVars, val string) error {
+		vars.TiDBTableCacheLease = tidbOptInt64(val, DefTiDBTableCacheLease)
+		return nil
+	}},
```

Comment:
I think this won't work correctly: It is `ScopeGlobal`, but when updating it will change the sessionVars of the current session only. So in the case that you are in another session, or don't explicitly 're-set' the value `vars.TiDBTableCacheLease` will not apply.

What you could do is:
1) Remove the `SetGlobal` func, and instead rely on using the globalvaraccessor to retrieve the expected value. This has the downside that you will always be casting from string to unsigned so you won't want to do this for hot code paths :(
2) Change the `SetGlobal` func to call some other server-wide func that you maintain. If you do this, make sure to also update `domain/sysvar_cache.go`. Currently we don't fire the 'SetGlobal` func on peers when a change happens, but we might in future:
 https://github.com/pingcap/tidb/blob/9ad00962f6561b25fe0fe5290ba16de1c3901ff7/domain/sysvar_cache.go#L162-L247

Reply:
OK, I choose 3

Reply:
OK, I choose 3

Reply:
Sounds good to me. I'll resolve this comment.